overlord/snapstate/handlers.go

```diff
@@ -1716,6 +1716,33 @@ func (m *SnapManager) doToggleSnapFlags(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// installModeDisabledServices returns what services with
+// "install-mode: disabled" should be disabled.
+//
+// This is done by checking if the service is new in this
+// snap revision and if it is, it is auto-disabled.
+func installModeDisabledServices(st *state.State, snapst *SnapState, currentInfo *snap.Info) (svcsToDisable []string, err error) {
+	prevCurrentSvcs := map[string]bool{}
+
+	if psi := snapst.previousSideInfo(); psi != nil {
+		var prevCurrentInfo *snap.Info
+		if prevCurrentInfo, err = Info(st, snapst.InstanceName(), psi.Revision); prevCurrentInfo != nil {
+			for _, prevSvc := range prevCurrentInfo.Services() {
+				prevCurrentSvcs[prevSvc.Name] = true
+			}
+		}
+	}
```

Comment:
I spent a good while pondering about why this is needed (and played around with removing this logic & running the spread test); this actually handles the case where we install the given service for the first time (which could be a refresh to a new revision), and otherwise we start+enable it if it got enabled by other means (such as `snap enable..`). Perhaps worth a comment.

Reply:
I added/updated the comment (and also picked the comment you suggested above.